front/app/containers/EventsPage/EventsViewer/index.tsx

```diff
@@ -59,26 +60,28 @@ interface Props {
   className?: string;
 }
 
-const EVENTS_PER_PAGE = 10;
-
 const EventsViewer = memo<Props>(
   ({ title, fallbackMessage, eventsTime, className }) => {
-    const [projectIds, setProjectIds] = useState<string[]>([]);
-    const [currentPageNumber, setCurrentPageNumber] = useState<number>(1);
-
-    const { events, lastPageNumber } = useEvents({
-      projectIds,
+    const {
+      events,
+      currentPage,
+      lastPage,
+      setProjectIds,
+      setCurrentPage,
+    } = useEvents({
       futureOnly: eventsTime === 'future',
       pastOnly: eventsTime === 'past',
-      pageNumber: currentPageNumber,
-      pageSize: EVENTS_PER_PAGE,
     });
 
+    const eventsLoading = isNil(events);
```

Comment:
Have you checked what events looks like when there are not events on the platform at all? Is it also `null`? Then it would keep spinning.

Reply:
It will return an empty array, resulting in the illustration and text saying that there are no events :) just tested to make sure